apisix/core/utils.lua

```diff
@@ -203,6 +209,26 @@ function _M.validate_header_value(value)
 end
 
 
+function _M.gethostname()
+    if hostname then
+        return hostname
+    end
+
+    local buf = get_string_buf(max_hostname_len)
+
+    if C.gethostname(buf, max_hostname_len) == 0 then
+        buf[max_hostname_len - 1] = str_byte('\0')
+        hostname = ffi_string(buf, ffi.C.strlen(buf))
+
+    else
+        log.alert("ffi.C.gethostname() failed")
```

Comment:
make a try:

use `io.popen("/bin/hostname")` to fetch the host name

Reply:
That'd be concise, i will try it.

Reply:
Changed.